src/main/kotlin/io/openfuture/snapshot/util/ThreadUtils.kt

```diff
@@ -0,0 +1,9 @@
+package io.openfuture.snapshot.util
+
+import java.util.concurrent.ExecutorService
+
+fun ExecutorService.awaitAll() {
+    this.shutdown()
```

Comment:
You would load CPU up to 100 percent without use sleep inside these loop. Do not do like this